apps/condo/domains/common/utils/import/date.js

```diff
@@ -107,16 +111,53 @@ function isDateStrValid (dateStr, formats = DEFAULT_DATE_PARSING_FORMATS) {
             return format.substring(0, format.length - offsetFromEnd)
         })
 
-        return dayjs(dateStr, offsetFormats, true).isValid()
+        return parseDayjs(dateStr, offsetFormats).isValid()
     }
 
     return false
 }
 
+/**
+ * @param {(dateString: string, formats: string | string[], strict: boolean) => dayjs.Dayjs} dayjsFunc
+ * @param dateString
+ * @param {({format: string, validate: (date: dayjs.Dayjs) => boolean} | string)[]} formats
+ * @param strict
+ * @returns dayjs.Dayjs
+ * */
+function tryParseDateWithRestrictions (dayjsFunc, dateString, formats, strict) {
```

Comment:
Its usually a bad practice for functions to have more than 3 args, consider grouping something into options...